utils/pattern/src/frontend/mod.rs

```diff
@@ -88,6 +88,16 @@ impl<B: PatternBackend> core::fmt::Debug for Pattern<B> {
     }
 }
 
+impl<B: PatternBackend> Default for Box<Pattern<B>>
+where
+    Box<B::Store>: Default,
+{
+    fn default() -> Self {
+        #[allow(clippy::unwrap_used)]
+        Pattern::try_from_items(core::iter::empty()).unwrap()
+    }
```

Comment:
Issue: If you need a way to generate an empty pattern, either:

1. Add a trait function, or
2. Add commentary on the trait that it must always succeed on construction from an empty iterator of items, and then cite that invariant here in the clippy::unwrap_used

Reply:
Do you not already allow for a pattern to be the empty string?

Reply:
I think all three backends currently allow it, but if you want to `unwrap()`, you should add it as a trait invariant. I don't think it is currently documented as such. Alternatively, add `new_empty() -> &'static Self::Store` to the trait.

Reply:
Done